llvm/test/Transforms/InstCombine/ptrtoint-nullgep.ll

```diff
@@ -16,8 +16,14 @@ declare void @use_i64(i64)
 declare void @use_ptr(ptr addrspace(1))
 
 define i64 @constant_fold_ptrtoint_gep_zero() {
-; ALL-LABEL: define {{[^@]+}}@constant_fold_ptrtoint_gep_zero() {
-; ALL-NEXT:    ret i64 0
+; LLPARSER-LABEL: define {{[^@]+}}@constant_fold_ptrtoint_gep_zero() {
+; LLPARSER-NEXT:    ret i64 ptrtoint (ptr addrspace(1) null to i64)
+;
+; INSTSIMPLIFY-LABEL: define {{[^@]+}}@constant_fold_ptrtoint_gep_zero() {
+; INSTSIMPLIFY-NEXT:    ret i64 ptrtoint (ptr addrspace(1) null to i64)
```

Comment:
Hmm, shouldn't we have the datalayout in instsimplify, does this mean we don't use the datalayout aware constant folding logic there?